src/calico_manifests.py

```diff
@@ -73,7 +73,9 @@ def __call__(self, obj) -> None:
         for container in containers:
             if container.name == "calico-node":
                 env = container.env
-                ipauto_env = EnvVar("IP_AUTODETECTION_METHOD", "skip-interface=lxd.*,fan.*")
+                ipauto_env = EnvVar(
+                    "IP_AUTODETECTION_METHOD", f"cidr={self.manifests.config.get('ipv4_cidr')}"
```

Comment:
where does this `ipv4_cidr` come from?

https://github.com/charmed-kubernetes/charm-calico/pull/104/files#diff-0ba5352c42dd9d68ded29429036dc5656e241df05c84918ba6d9a1d2b12dbfc2L351-L355

Reply:
`ipv4_cidr` is sourced from the `cni_config` dict. You can see the update on this [line](https://github.com/charmed-kubernetes/charm-calico/pull/104/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR402)